processing/src/main/java/org/apache/druid/segment/filter/OrFilter.java

```diff
@@ -71,6 +75,167 @@ public OrFilter(List<Filter> filters)
     this(new LinkedHashSet<>(filters));
   }
 
+  @Override
+  public <T> FilterBundle makeFilterBundle(
+      ColumnIndexSelector columnIndexSelector,
+      BitmapResultFactory<T> bitmapResultFactory,
+      int selectionRowCount,
+      int totalRowCount,
+      boolean includeUnknown,
+      boolean allowPartialIndex
+  )
+  {
+    final List<T> indexes = new ArrayList<>();
+    final List<Filter> indexFilters = new ArrayList<>();
+    final List<Filter> matcherFilters = new ArrayList<>();
+
+    final long bitmapConstructionStartNs = System.nanoTime();
+    for (Filter subfilter : filters) {
+      boolean needMatcher = true;
+      final BitmapColumnIndex index = subfilter.getBitmapColumnIndex(columnIndexSelector);
```

Comment:
I was wondering if we can delegate to child's filter bundles, and have a way of combining the filter bundles?

Reply:
Fwiw, it should be possible to convert a bitmap index into a matcher by testing the id of the row against the bitmap.  It should still be better than loading the value and grabbing the matcher and you can combine it together with a matcher to get the correct semantics inside of the OR here for the times when you need to do it all via a matcher

Reply:
The code is actually doing that, if not all of the sub-filter can be converted into indexes, we re-use them to make a matcher, i'll add some comments and such to try to make it clearer

Reply:
i did end up doing this to make OrFilter delegate to child bundles. Its a bit more complicated than the initial implementation, but the code comments should explain the possible outcomes for different shapes of child bundles